torch/testing/_internal/common_methods_invocations.py

```diff
@@ -11897,6 +11897,7 @@ def ref_pairwise_distance(input1, input2):
         supports_out=False,
         supports_forward_ad=True,
         promotes_int_to_int64=True,
+        supports_sparse=True,
```

Comment:
This is very neat! 
The only problem is that `dtype` is not a standard kwarg for a reduction operation (cf. the definition of `ReductionOpInfo`). As such, you'll need to create another `ReductionOpInfo` that looks just like this one, but also passes a set `dtype` to its `kwargs` to be able to test that everything works as expected.

Reply:
Looking at the optinos for this OpInfo, I believe it should also have the flag `result_dtype=True`, right @mruberry ?
Now, it's not clear to me whether this + `supports_sparse` would test everything for sparse tensors. In particular, I'm thinking of backward + `dtype=` (which I think it may break at the moment).

Reply:
Setting `result_dtype=True` is incompatible with `promotes_int_to_int64=True`.

Reply:
@Lezcano can you give me an example test we'd want to hit?

Reply:
One where we differentiate the an expression that uses an explicit `dtype=`.